internal/controller/common/numaflowtypes/pipeline.go

```diff
@@ -198,14 +198,30 @@ func PipelineWithoutDesiredPhase(pipeline *unstructured.Unstructured) {
 	}
 }
 
-func PipelineWithoutScaleMinMax(pipeline *unstructured.Unstructured) {
-	unstructured.RemoveNestedField(pipeline.Object, "spec", "scale", "min")
-	unstructured.RemoveNestedField(pipeline.Object, "spec", "scale", "max")
+func PipelineWithoutScaleMinMax(pipeline *unstructured.Unstructured) error {
+	// for each Vertex, remove the scale min and max:
+	vertices, _, _ := unstructured.NestedSlice(pipeline.Object, "spec", "vertices")
 
-	// if "scale" is there and empty, remove it
-	spec := pipeline.Object["spec"].(map[string]interface{})
-	scaleMap, found := spec["scale"].(map[string]interface{})
-	if found && len(scaleMap) == 0 {
-		unstructured.RemoveNestedField(pipeline.Object, "spec", "scale")
+	modifiedVertices := make([]interface{}, 0)
+	for _, vertex := range vertices {
+		if vertexAsMap, ok := vertex.(map[string]any); ok {
+
+			unstructured.RemoveNestedField(vertexAsMap, "scale", "min")
+			unstructured.RemoveNestedField(vertexAsMap, "scale", "max")
+
+			// if "scale" is there and empty, remove it
+			scaleMap, found := vertexAsMap["scale"].(map[string]interface{})
+			if found && len(scaleMap) == 0 {
+				unstructured.RemoveNestedField(vertexAsMap, "scale")
+			}
+
+			modifiedVertices = append(modifiedVertices, vertexAsMap)
+		}
 	}
+
+	err := unstructured.SetNestedSlice(pipeline.Object, modifiedVertices, "spec", "vertices")
+	if err != nil {
+		return err
+	}
+	return nil
```

Comment:
this was a total goof on my previous PR when I was going a little too fast. This code hadn't been used and tested at the time, which was why it got in there.

Reply:
Note that I have a unit test which is checking this now - https://github.com/numaproj/numaplane/blob/pipeline-scale-0/internal/controller/pipelinerollout/pipelinerollout_controller_test.go#L532-L545